compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -513,6 +513,34 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
       Match(param, cases)
     }
 
+  /** For an enum T:
+   *
+   *     def enumLabel(x: MirroredMonoType) = x.enumLabel
+   *
+   *  For  sealed trait with children of normalized types C_1, ..., C_n:
+   *
+   *     def enumLabel(x: MirroredMonoType) = x match {
+   *        case _: C_1 => "C_1"
+   *        ...
+   *        case _: C_n => "C_n"
+   *     }
+   *
+   *  Here, the normalized type of a class C is C[?, ...., ?] with
+   *  a wildcard for each type parameter. The normalized type of an object
+   *  O is O.type.
+   */
+  def enumLabelBody(cls: Symbol, param: Tree)(using Context): Tree =
+    if (cls.is(Enum)) param.select(nme.enumLabel).ensureApplied
+    else {
+      val cases =
+        for ((child, idx) <- cls.children.zipWithIndex) yield {
+          val patType = if (child.isTerm) child.termRef else child.rawTypeRef
+          val pat = Typed(untpd.Ident(nme.WILDCARD).withType(patType), TypeTree(patType))
+          CaseDef(pat, EmptyTree, Literal(Constant(child.name.toString)))
+        }
+      Match(param, cases)
+    }
```

Comment:
Why do we need this? I believe there are other way to get to the names of enum constants.

Generally, we have to be very careful not to generate too much unnecessary code for this. Every single step looks reasonable until we hit a limit where people will get turned off since it's just too much code . So , every method we generate has to be carefully justified and the alternatives have to be weighed up against the addition.

Reply:
This is a good point and can definitely be removed from this PR.

The main issue was that with scala.Enum trait removed there would be no generic way to reflect on the label of some enum, unless you rely on productPrefix which can be overloaded

Reply:
this has been removed to reconsider later